### DISCUSSION TITLE
Does eslint support share the config contains overrides？

### DISCUSSION THREAD

**Discussion by mrdulin** [eslint/eslint#18565] [Q&A]
I try to create a share config for my project. 

`peerDependencies` in `eslint-config-myconfig/package.json`:
```json
  "peerDependencies": {
    "eslint": ">=8.3.0"
  }
```

`eslint-config-myconfig/index.js`:
```js
module.exports = {
  root: true,
  env: {
    node: true,
    es6: true,
  },
  parserOptions: {
    ecmaVersion: 8,
  },
  extends: [
    "eslint:recommended",
    "plugin:import/errors",
    "plugin:import/warnings",
    "plugin:import/typescript",
    "plugin:prettier/recommended",
  ],
  rules: {
    "prettier/prettier": [
      "error",
      { endOfLine: "auto" },
      {
        usePrettierrc: true,
      },
    ],
    "import/no-named-as-default": "off",
    "import/order": [
      "error",
      {
        alphabetize: {
          order: "asc",
          caseInsensitive: true,
        },
      },
    ],
  },
  overrides: [
    {
      files: ["**/*.ts", "**/*.tsx"],
      parser: "@typescript-eslint/parser",
      settings: {
        react: {
          version: "detect",
        },
      },
      parserOptions: {
        sourceType: "module",
        project: "./tsconfig.json",
      },
      env: {
        browser: true,
      },
      extends: [
        "plugin:@typescript-eslint/recommended",
        "plugin:react/recommended",
        "plugin:react-hooks/recommended",
        "plugin:jsx-a11y/recommended",
      ],
      rules: {
        "@typescript-eslint/no-unused-vars": ["error"]
      },
    },
  ],
};
```

`project/eslintrc.js`:
```js
module.exports = {
  root: true,
  extends: ["eslint-config-dm"],
  overrides: [
    {
       files: ["**/*.ts", "**/*.tsx"],
       parserOptions: {
         sourceType: "module",
         project: "./tsconfig.json",
      },
    }
  ]
};
```

There is a `project/tsconfig.json` file in project root directory.

Run `npm link` command. Restart Eslint Server, unused variable code:
```ts
const colors = ["red", , "blue"];
```

But eslint does not report the error. It should report:

> 'colors' is assigned a value but never used.

And,

> Unexpected comma in middle of array.

I expect the `overrides` config of `project/eslintrc.js` should be merged with `overrides` config of  `eslint-config-myconfig` package. Is this possible? 

The package repo: https://github.com/mrdulin/eslint-config-dm/blob/main/index.js

**Comment by Rec0iL99** [comment]
Hi @mrdulin, you can check the final computed ESLint config file used for linting by using the [`--print-config`](https://eslint.org/docs/latest/use/command-line-interface#--print-config) CLI option.

**Reply by mrdulin** [reply]
This CLI option is helpful. But the `overrides` config is not there.

<details>
<summary>Output</summary>
<pre>
$ npx eslint --print-config ./eslintrc.js
{
  "env": {
    "node": true,
    "es6": true
  },
  "globals": {},
  "parser": null,
  "parserOptions": {
    "ecmaVersion": 8
  },
  "plugins": [
    "import",
    "prettier"
  ],
  "rules": {
    "prettier/prettier": [
      "error",
      {
        "endOfLine": "auto"
      },
      {
        "usePrettierrc": true
      }
    ],
    "import/no-named-as-default": [
      "off"
    ],
    "import/order": [
      "error",
      {
        "alphabetize": {
          "order": "asc",
          "caseInsensitive": true
        },
        "warnOnUnassignedImports": false
      }
    ],
    "arrow-body-style": [
      "off"
    ],
    "prefer-arrow-callback": [
      "off"
    ],
    "curly": [
      0
    ],
    "lines-around-comment": [
      0
    ],
    "max-len": [
      0
    ],
    "no-confusing-arrow": [
      0
    ],
    "no-mixed-operators": [
      0
    ],
    "no-tabs": [
      0
    ],
    "no-unexpected-multiline": [
      0
    ],
    "quotes": [
      0
    ],
    "@typescript-eslint/quotes": [
      0
    ],
    "babel/quotes": [
      0
    ],
    "vue/html-self-closing": [
      0
    ],
    "vue/max-len": [
      0
    ],
    "array-bracket-newline": [
      "off"
    ],
    "array-bracket-spacing": [
      "off"
    ],
    "array-element-newline": [
      "off"
    ],
    "arrow-parens": [
      "off"
    ],
    "arrow-spacing": [
      "off"
    ],
    "block-spacing": [
      "off"
    ],
    "brace-style": [
      "off"
    ],
    "comma-dangle": [
      "off"
    ],
    "comma-spacing": [
      "off"
    ],
    "comma-style": [
      "off"
    ],
    "computed-property-spacing": [
      "off"
    ],
    "dot-location": [
      "off"
    ],
    "eol-last": [
      "off"
    ],
    "func-call-spacing": [
      "off"
    ],
    "function-call-argument-newline": [
      "off"
    ],
    "function-paren-newline": [
      "off"
    ],
    "generator-star": [
      "off"
    ],
    "generator-star-spacing": [
      "off"
    ],
    "implicit-arrow-linebreak": [
      "off"
    ],
    "indent": [
      "off"
    ],
    "jsx-quotes": [
      "off"
    ],
    "key-spacing": [
      "off"
    ],
    "keyword-spacing": [
      "off"
    ],
    "linebreak-style": [
      "off"
    ],
    "multiline-ternary": [
      "off"
    ],
    "newline-per-chained-call": [
      "off"
    ],
    "new-parens": [
      "off"
    ],
    "no-arrow-condition": [
      "off"
    ],
    "no-comma-dangle": [
      "off"
    ],
    "no-extra-parens": [
      "off"
    ],
    "no-extra-semi": [
      "off"
    ],
    "no-floating-decimal": [
      "off"
    ],
    "no-mixed-spaces-and-tabs": [
      "off"
    ],
    "no-multi-spaces": [
      "off"
    ],
    "no-multiple-empty-lines": [
      "off"
    ],
    "no-reserved-keys": [
      "off"
    ],
    "no-space-before-semi": [
      "off"
    ],
    "no-trailing-spaces": [
      "off"
    ],
    "no-whitespace-before-property": [
      "off"
    ],
    "no-wrap-func": [
      "off"
    ],
    "nonblock-statement-body-position": [
      "off"
    ],
    "object-curly-newline": [
      "off"
    ],
    "object-curly-spacing": [
      "off"
    ],
    "object-property-newline": [
      "off"
    ],
    "one-var-declaration-per-line": [
      "off"
    ],
    "operator-linebreak": [
      "off"
    ],
    "padded-blocks": [
      "off"
    ],
    "quote-props": [
      "off"
    ],
    "rest-spread-spacing": [
      "off"
    ],
    "semi": [
      "off"
    ],
    "semi-spacing": [
      "off"
    ],
    "semi-style": [
      "off"
    ],
    "space-after-function-name": [
      "off"
    ],
    "space-after-keywords": [
      "off"
    ],
    "space-before-blocks": [
      "off"
    ],
    "space-before-function-paren": [
      "off"
    ],
    "space-before-function-parentheses": [
      "off"
    ],
    "space-before-keywords": [
      "off"
    ],
    "space-in-brackets": [
      "off"
    ],
    "space-in-parens": [
      "off"
    ],
    "space-infix-ops": [
      "off"
    ],
    "space-return-throw-case": [
      "off"
    ],
    "space-unary-ops": [
      "off"
    ],
    "space-unary-word-ops": [
      "off"
    ],
    "switch-colon-spacing": [
      "off"
    ],
    "template-curly-spacing": [
      "off"
    ],
    "template-tag-spacing": [
      "off"
    ],
    "unicode-bom": [
      "off"
    ],
    "wrap-iife": [
      "off"
    ],
    "wrap-regex": [
      "off"
    ],
    "yield-star-spacing": [
      "off"
    ],
    "@babel/object-curly-spacing": [
      "off"
    ],
    "@babel/semi": [
      "off"
    ],
    "@typescript-eslint/brace-style": [
      "off"
    ],
    "@typescript-eslint/comma-dangle": [
      "off"
    ],
    "@typescript-eslint/comma-spacing": [
      "off"
    ],
    "@typescript-eslint/func-call-spacing": [
      "off"
    ],
    "@typescript-eslint/indent": [
      "off"
    ],
    "@typescript-eslint/keyword-spacing": [
      "off"
    ],
    "@typescript-eslint/member-delimiter-style": [
      "off"
    ],
    "@typescript-eslint/no-extra-parens": [
      "off"
    ],
    "@typescript-eslint/no-extra-semi": [
      "off"
    ],
    "@typescript-eslint/object-curly-spacing": [
      "off"
    ],
    "@typescript-eslint/semi": [
      "off"
    ],
    "@typescript-eslint/space-before-function-paren": [
      "off"
    ],
    "@typescript-eslint/space-infix-ops": [
      "off"
    ],
    "@typescript-eslint/type-annotation-spacing": [
      "off"
    ],
    "babel/object-curly-spacing": [
      "off"
    ],
    "babel/semi": [
      "off"
    ],
    "flowtype/boolean-style": [
      "off"
    ],
    "flowtype/delimiter-dangle": [
      "off"
    ],
    "flowtype/generic-spacing": [
      "off"
    ],
    "flowtype/object-type-curly-spacing": [
      "off"
    ],
    "flowtype/object-type-delimiter": [
      "off"
    ],
    "flowtype/quotes": [
      "off"
    ],
    "flowtype/semi": [
      "off"
    ],
    "flowtype/space-after-type-colon": [
      "off"
    ],
    "flowtype/space-before-generic-bracket": [
      "off"
    ],
    "flowtype/space-before-type-colon": [
      "off"
    ],
    "flowtype/union-intersection-spacing": [
      "off"
    ],
    "react/jsx-child-element-spacing": [
      "off"
    ],
    "react/jsx-closing-bracket-location": [
      "off"
    ],
    "react/jsx-closing-tag-location": [
      "off"
    ],
    "react/jsx-curly-newline": [
      "off"
    ],
    "react/jsx-curly-spacing": [
      "off"
    ],
    "react/jsx-equals-spacing": [
      "off"
    ],
    "react/jsx-first-prop-new-line": [
      "off"
    ],
    "react/jsx-indent": [
      "off"
    ],
    "react/jsx-indent-props": [
      "off"
    ],
    "react/jsx-max-props-per-line": [
      "off"
    ],
    "react/jsx-newline": [
      "off"
    ],
    "react/jsx-one-expression-per-line": [
      "off"
    ],
    "react/jsx-props-no-multi-spaces": [
      "off"
    ],
    "react/jsx-tag-spacing": [
      "off"
    ],
    "react/jsx-wrap-multilines": [
      "off"
    ],
    "standard/array-bracket-even-spacing": [
      "off"
    ],
    "standard/computed-property-even-spacing": [
      "off"
    ],
    "standard/object-curly-even-spacing": [
      "off"
    ],
    "unicorn/empty-brace-spaces": [
      "off"
    ],
    "unicorn/no-nested-ternary": [
      "off"
    ],
    "unicorn/number-literal-case": [
      "off"
    ],
    "vue/array-bracket-newline": [
      "off"
    ],
    "vue/array-bracket-spacing": [
      "off"
    ],
    "vue/arrow-spacing": [
      "off"
    ],
    "vue/block-spacing": [
      "off"
    ],
    "vue/block-tag-newline": [
      "off"
    ],
    "vue/brace-style": [
      "off"
    ],
    "vue/comma-dangle": [
      "off"
    ],
    "vue/comma-spacing": [
      "off"
    ],
    "vue/comma-style": [
      "off"
    ],
    "vue/dot-location": [
      "off"
    ],
    "vue/func-call-spacing": [
      "off"
    ],
    "vue/html-closing-bracket-newline": [
      "off"
    ],
    "vue/html-closing-bracket-spacing": [
      "off"
    ],
    "vue/html-end-tags": [
      "off"
    ],
    "vue/html-indent": [
      "off"
    ],
    "vue/html-quotes": [
      "off"
    ],
    "vue/key-spacing": [
      "off"
    ],
    "vue/keyword-spacing": [
      "off"
    ],
    "vue/max-attributes-per-line": [
      "off"
    ],
    "vue/multiline-html-element-content-newline": [
      "off"
    ],
    "vue/mustache-interpolation-spacing": [
      "off"
    ],
    "vue/no-extra-parens": [
      "off"
    ],
    "vue/no-multi-spaces": [
      "off"
    ],
    "vue/no-spaces-around-equal-signs-in-attribute": [
      "off"
    ],
    "vue/object-curly-newline": [
      "off"
    ],
    "vue/object-curly-spacing": [
      "off"
    ],
    "vue/object-property-newline": [
      "off"
    ],
    "vue/operator-linebreak": [
      "off"
    ],
    "vue/script-indent": [
      "off"
    ],
    "vue/singleline-html-element-content-newline": [
      "off"
    ],
    "vue/space-in-parens": [
      "off"
    ],
    "vue/space-infix-ops": [
      "off"
    ],
    "vue/space-unary-ops": [
      "off"
    ],
    "vue/template-curly-spacing": [
      "off"
    ],
    "indent-legacy": [
      "off"
    ],
    "no-spaced-func": [
      "off"
    ],
    "react/jsx-space-before-closing": [
      "off"
    ],
    "import/named": [
      "off"
    ],
    "import/no-named-as-default-member": [
      1
    ],
    "import/no-duplicates": [
      1
    ],
    "import/no-unresolved": [
      2
    ],
    "import/namespace": [
      2
    ],
    "import/default": [
      2
    ],
    "import/export": [
      2
    ],
    "constructor-super": [
      "error"
    ],
    "for-direction": [
      "error"
    ],
    "getter-return": [
      "error"
    ],
    "no-async-promise-executor": [
      "error"
    ],
    "no-case-declarations": [
      "error"
    ],
    "no-class-assign": [
      "error"
    ],
    "no-compare-neg-zero": [
      "error"
    ],
    "no-cond-assign": [
      "error"
    ],
    "no-const-assign": [
      "error"
    ],
    "no-constant-condition": [
      "error"
    ],
    "no-control-regex": [
      "error"
    ],
    "no-debugger": [
      "error"
    ],
    "no-delete-var": [
      "error"
    ],
    "no-dupe-args": [
      "error"
    ],
    "no-dupe-class-members": [
      "error"
    ],
    "no-dupe-else-if": [
      "error"
    ],
    "no-dupe-keys": [
      "error"
    ],
    "no-duplicate-case": [
      "error"
    ],
    "no-empty": [
      "error"
    ],
    "no-empty-character-class": [
      "error"
    ],
    "no-empty-pattern": [
      "error"
    ],
    "no-ex-assign": [
      "error"
    ],
    "no-extra-boolean-cast": [
      "error"
    ],
    "no-fallthrough": [
      "error"
    ],
    "no-func-assign": [
      "error"
    ],
    "no-global-assign": [
      "error"
    ],
    "no-import-assign": [
      "error"
    ],
    "no-inner-declarations": [
      "error"
    ],
    "no-invalid-regexp": [
      "error"
    ],
    "no-irregular-whitespace": [
      "error"
    ],
    "no-loss-of-precision": [
      "error"
    ],
    "no-misleading-character-class": [
      "error"
    ],
    "no-new-symbol": [
      "error"
    ],
    "no-nonoctal-decimal-escape": [
      "error"
    ],
    "no-obj-calls": [
      "error"
    ],
    "no-octal": [
      "error"
    ],
    "no-prototype-builtins": [
      "error"
    ],
    "no-redeclare": [
      "error"
    ],
    "no-regex-spaces": [
      "error"
    ],
    "no-self-assign": [
      "error"
    ],
    "no-setter-return": [
      "error"
    ],
    "no-shadow-restricted-names": [
      "error"
    ],
    "no-sparse-arrays": [
      "error"
    ],
    "no-this-before-super": [
      "error"
    ],
    "no-undef": [
      "error"
    ],
    "no-unreachable": [
      "error"
    ],
    "no-unsafe-finally": [
      "error"
    ],
    "no-unsafe-negation": [
      "error"
    ],
    "no-unsafe-optional-chaining": [
      "error"
    ],
    "no-unused-labels": [
      "error"
    ],
    "no-unused-vars": [
      "error"
    ],
    "no-useless-backreference": [
      "error"
    ],
    "no-useless-catch": [
      "error"
    ],
    "no-useless-escape": [
      "error"
    ],
    "no-with": [
      "error"
    ],
    "require-yield": [
      "error"
    ],
    "use-isnan": [
      "error"
    ],
    "valid-typeof": [
      "error"
    ]
  },
  "settings": {
    "import/extensions": [
      ".ts",
      ".tsx",
      ".js",
      ".jsx"
    ],
    "import/external-module-folders": [
      "node_modules",
      "node_modules/@types"
    ],
    "import/parsers": {
      "@typescript-eslint/parser": [
        ".ts",
        ".tsx"
      ]
    },
    "import/resolver": {
      "node": {
        "extensions": [
          ".ts",
          ".tsx",
          ".js",
          ".jsx"
        ]
      }
    }
  },
  "ignorePatterns": [
    "node_modules/*",
    "dist/*",
    "bin/",
    "libs/*",
    "!.prettierrc.js",
    "!.lintstagedrc.js"
  ]
}
</pre>
</details>

**Reply by mrdulin** [reply]
The `overrides` config merged correctly. The argument for the `--print-config` should be a file to be lint, not the `.eslintrc.js` itself.

e.g.
```bash
npx eslint --print-config src/Page.tsx
```

Besides, I didn't install the `typescript` as `peerDependencies` for `eslint-config-myconfig`
